scripts/install

```diff
@@ -0,0 +1,249 @@
+}
+
+_install_macos_prompt_to_update_path() {
+  INSTALL_PREFIX=$1
+
+  BIN_DIR="${INSTALL_PREFIX}/bin"
+
+  path_presented=false
+  path_set=false
+
+  if [ -w  "${HOME}" ]; then
+    if [ ! -e "${HOME}/.zprofile" ] || [ -w "${HOME}/.zprofile" ];then
+      printf "Add %s to PATH via .zprofile (Y/n)? " "${BIN_DIR}"
+      path_presented=true
+      read -r answer_zprofile
+      if [ "${answer_zprofile}" != "${answer_zprofile#[Nn]}" ];then
+        echo 'Not updating .zprofile'
+      else
+        _install_macos_add_to_profile "${BIN_DIR}" ".zprofile"
+        path_set=true
+      fi
+    fi
+
+    if [ ! -e "${HOME}/.profile" ] || [ -w "${HOME}/.profile" ];then
+      printf "Add %s to PATH via .profile (Y/n)? " "${BIN_DIR}"
+      path_presented=true
+      read -r answer_profile
+      if [ "${answer_profile}" != "${answer_profile#[Nn]}" ];then
+        echo 'Not updating .profile'
+      else
+        _install_macos_add_to_profile "${BIN_DIR}" ".profile"
+        path_set=true
+      fi
+    fi
+  fi
+  if ! "${path_presented}" || ! "${path_set}"; then
+    if ! "${path_presented}";then
+      echo "We couldn't write to .profile or .zprofile."
+    fi
+    echo "You should add ${BIN_DIR} to your PATH."
+  fi
+  echo ''
+}
+
+_install_macos_extract_files() {
+  TMP_DIR=$1
+  TMP_FILE=$2
+  INSTALL_PREFIX=$3
+
+  cd "${TMP_DIR}"
+  bsdtar xzf "${TMP_FILE}"
+  echo "Extracted outer tarball in-place"
+  if ! [ -d "${INSTALL_PREFIX}" ];then 
+    echo "Creating ${INSTALL_PREFIX}"
+    mkdir -p "${INSTALL_PREFIX}"
+  fi
+  echo "Created dir ${INSTALL_PREFIX}"
+  bsdtar -xzf Payload -C "${INSTALL_PREFIX}" opt
+  bsdtar -xzf Payload -C "${INSTALL_PREFIX}" --strip-components=3 usr/local/bin
+  echo "Extracted inner tarball to ${INSTALL_PREFIX}"
+  echo ''
+}
+
+_install_macos_download_files() {
+  ARCHIVE=$1
+  TMP_FILE=$2
+
+  echo "Downloading ${ARCHIVE}"
+  curl -s "${ARCHIVE}" -o "${TMP_FILE}"
+  echo "Downloaded at ${TMP_FILE}"
+}
+
+install_macos() {
+  BUCKET_URL=$1
+  VERSION=$2
+  INSTALL_PREFIX=$3
+
+  FILENAME="imandrax-macos-aarch64-${VERSION}.pkg"
+  ARCHIVE="${BUCKET_URL}/${FILENAME}"
+  TMP_DIR="${TMPDIR:-/tmp}"
+  TMP_FILE="${TMP_DIR}${FILENAME}"
+
+  _install_macos_download_files "${ARCHIVE}" "${TMP_FILE}"
+  _install_macos_extract_files "${TMP_DIR}" "${TMP_FILE}" "${INSTALL_PREFIX}"
+
+  # modify executable to find libs
+  sed -i'.backup' "s#DIR=/opt/imandrax#DIR=${INSTALL_PREFIX}/opt/imandrax#" \
+    "${INSTALL_PREFIX}/bin/imandrax-cli"
+  
+  # clean up temp files
+  rm -rf "${INSTALL_PREFIX}/bin/imandrax-cli.backup"
+
+  _install_macos_check_files_present "${INSTALL_PREFIX}"
+  _install_macos_prompt_to_update_path "${INSTALL_PREFIX}"
+
+  CONFIG_DIR="${HOME}/.config/imandrax"
+
+  _install_macos_prompt_for_api_key "${CONFIG_DIR}"
+
+  
+  cat << EOF
+***********************
+* Installed ImandraX! *
+***********************
+
+EOF
+
+  echo 'See the docs for more info: '
+  echo 'https://github.com/imandra-ai/imandrax-api'
```

Comment:
If you're already paying the complexity-cost of a here-doc, I think you might as well collapse these two echos into it.